src/components/SetupTab/ArgInput/index.jsx

```diff
@@ -8,159 +8,223 @@ import Row from 'react-bootstrap/Row';
 import InputGroup from 'react-bootstrap/InputGroup';
 import Modal from 'react-bootstrap/Modal';
 
+function FormLabel(props) {
+  return (
+    <Form.Label column sm="3" htmlFor={props.argkey}>
+      {props.children}
+    </Form.Label>
+  );
+}
+FormLabel.propTypes = {
+  argkey: PropTypes.string.isRequired,
+  children: PropTypes.element.isRequired,
+};
 
-export class ArgInput extends React.PureComponent {
-  constructor(props) {
-    super(props)
-  }
+function Feedback(props) {
+  return (
+    <Form.Control.Feedback type="invalid" id={`${props.argkey}-feedback`}>
+      {`${props.argtype} : ${(props.message)}`}
+    </Form.Control.Feedback>
+  );
+}
+Feedback.propTypes = {
+  argkey: PropTypes.string.isRequired,
+  argtype: PropTypes.string.isRequired,
+  message: PropTypes.string,
+};
+Feedback.defaultProps = {
+  message: '',
+};
 
+export default class ArgInput extends React.PureComponent {
   render() {
-    const argkey = this.props.argkey
-    const argSpec = this.props.argSpec
-    let ArgInput;
+    const {
+      argkey,
+      argSpec,
+      handleBoolChange,
+      handleChange,
+      isValid,
+      selectFile,
+      touched,
+      ui_option,
+      validationMessage,
+      value,
+    } = this.props;
```

Comment:
I've noticed this new kind of set up in a few files. Is there anything you can comment on that might be interesting to me from a future developer point of view?

Reply:
This is [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment). When it's done with `this.props` in a `render()` method, it's mainly about readability and avoiding all the extra `this.props` prefixes that would otherwise be needed to refer to each of these objects.

But destructring comes up in lots of other useful ways too, with examples in the MDN docs.